src/Evolution/Systems/GeneralizedHarmonic/Constraints.cpp

```diff
@@ -467,6 +469,76 @@ void four_index_constraint(
     }
   }
 }
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+Scalar<DataType> constraint_energy(
+    const tnsr::a<DataType, SpatialDim, Frame>& gauge_constraint,
+    const tnsr::a<DataType, SpatialDim, Frame>& f_constraint,
+    const tnsr::ia<DataType, SpatialDim, Frame>& two_index_constraint,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& three_index_constraint,
+    const tnsr::ijaa<DataType, SpatialDim, Frame>& four_index_constraint,
+    const tnsr::II<DataType, SpatialDim, Frame>& inverse_spatial_metric,
+    double gauge_constraint_multiplier, double two_index_constraint_multiplier,
+    double three_index_constraint_multiplier,
+    double four_index_constraint_multiplier) noexcept {
+  auto energy = make_with_value<Scalar<DataType>>(gauge_constraint, 0.0);
+  constraint_energy<SpatialDim, Frame, DataType>(
+      &energy, gauge_constraint, f_constraint, two_index_constraint,
+      three_index_constraint, four_index_constraint, inverse_spatial_metric,
+      gauge_constraint_multiplier, two_index_constraint_multiplier,
+      three_index_constraint_multiplier, four_index_constraint_multiplier);
+  return energy;
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+void constraint_energy(
+    gsl::not_null<Scalar<DataType>*> energy,
+    const tnsr::a<DataType, SpatialDim, Frame>& gauge_constraint,
+    const tnsr::a<DataType, SpatialDim, Frame>& f_constraint,
+    const tnsr::ia<DataType, SpatialDim, Frame>& two_index_constraint,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& three_index_constraint,
+    const tnsr::ijaa<DataType, SpatialDim, Frame>& four_index_constraint,
+    const tnsr::II<DataType, SpatialDim, Frame>& inverse_spatial_metric,
+    double gauge_constraint_multiplier, double two_index_constraint_multiplier,
+    double three_index_constraint_multiplier,
+    double four_index_constraint_multiplier) noexcept {
+  if (get_size(get(*energy)) != get_size(get<0>(f_constraint))) {
+    *energy = Scalar<DataType>{get<0>(f_constraint)};
```

Comment:
Just set it to the size, don't bother copying the data

Reply:
Never mind...I'll just use make_with_value

Reply:
No, don't use `make_with_value`, just do `Scalar<DataType>{get_size(get<0>(f_constraint))}`

Reply:
Thanks @nilsdeppe! I didn't realize you could resize like that.